### PR TITLE
Adjust snapshot index resolution behavior to be more intuitive

### LINKED ISSUE #78320: Cannot restore snapshot on new cluster
**Elasticsearch version** (`bin/elasticsearch --version`): Version: 7.14.1, Build: default/docker/66b55ebfa59c92c15db3f69a335d500018b3331e/2021-08-26T09:01:05.390870785Z, JVM: 16.0.2

**Plugins installed**: []

**JVM version** (`java -version`): OpenJDK 64-Bit Server VM Temurin-16.0.2+7 (build 16.0.2+7, mixed mode, sharing)

**OS version** (`uname -a` if on a Unix-like system): Linux d3463a9ac7de 4.9.0-14-amd64 #1 SMP Debian 4.9.246-2 (2020-12-17) x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:

Trying to restore snapshot on a new single node cluster throws error:
```
"type" : "snapshot_restore_exception",
"reason" : "[backup:snapshot-2021.09.23/esJtA1MeRcenJbz3tkIL2A] cannot restore index [.geoip_databases] because an open index with same name already exists in the cluster. Either close or delete the existing index or restore the index under a different name by providing a rename pattern and replacement name"
```

**Steps to reproduce**:

 1. `PUT /_snapshot/backup/%3Csnapshot-%7Bnow%2Fd%7D%3E`
 2. Create new single node cluster
 3. `POST /_snapshot/backup/snapshot-2021.09.23/_restore`

**Provide logs (if relevant)**:

Discuss url: https://discuss.elastic.co/t/cannot-restore-snapshot-to-new-single-node-cluster/285025

### PR DESCRIPTION
**Pull request by AthenaEryma**
This PR removes the logic used to ensure backwards-compatible behavior
in the Snapshot and Restore APIs and updates the tests to check the new
logic. This is technically a breaking change, but significantly improves
the usability of snapshots in the presence of system indices.

The new behavior is as follows:
- When creating a snapshot, system indices are not resolved based on the
`indices` field in the request. System indices are included or excluded
using the `include_global_state` or `feature_states` fields, defaulting
to snapshotting all system indices.
- When restoring a snapshot, system indices cannot be requested using
the `indices` field in the request. System indices are included or
excluded using the `include_global_state` or `feature_states` fields,
defaulting to restoring no system indices.

Fixes https://github.com/elastic/elasticsearch/issues/78320

---

It's worth noting that this is a breaking change, but we deprecated the behavior that won't work anymore a while ago, and I don't think anyone likes the current behavior - it's a mess because of shenanigans we had to pull to maintain BWC.

### CODE REVIEW

**Review by williamrandolph — CHANGES_REQUESTED** [review]
Production code changes look straightforward. I only have a couple of cleanup nitpicks and suggestions, and it looks like there might be a nontrivial test failure to fix.

File: server/src/internalClusterTest/java/org/elasticsearch/snapshots/SystemIndicesSnapshotIT.java [hunk 1]
```diff
@@ -513,24 +482,13 @@ public void testRestoreSystemIndicesAsGlobalStateWithEmptyListOfFeatureStatesNoI
...
-                    + "] because an open index with same name already exists in the cluster."
-            )
-        );
+        RestoreSnapshotResponse restoreResponse = clusterAdmin().prepareRestoreSnapshot(REPO_NAME, "test-snap")
+            .setWaitForCompletion(true)
+            .setRestoreGlobalState(true)
+            .setFeatureStates(new String[] { randomFrom("none", "NONE") })
+            .get();
+        assertThat(restoreResponse.getRestoreInfo().successfulShards(), equalTo(restoreResponse.getRestoreInfo().successfulShards()));
+        // GWB> What do we check here?
```
[the remark this answers is not in the document]
Comment (gwbrown) [reply server/src/internalClusterTest/java/org/elasticsearch/snapshots/SystemIndicesSnapshotIT.java]: This test case was now a duplicate of another! So I deleted it.

**Review by williamrandolph — APPROVED** [review]
LGTM!

**Comment by gwbrown** [comment]
@elasticmachine update branch

**Review by original-brownbear — APPROVED** [review]
LGTM from my end :) Sorry for the delay here!

**Pull request merged**